applications/luci-app-ddns/htdocs/luci-static/resources/view/ddns/overview.js

```diff
@@ -0,0 +1,993 @@
+			section_id, cfg_detail_ip, cfg_update, cfg_status, host, ip, last_update, 
+			next_update, service_status, reload, cfg_enabled, stop,
+			ddns_enabled = map.querySelector('[data-name="_enabled"]').querySelector('.cbi-value-field'),
+			ddns_toggle = map.querySelector('[data-name="_toggle"]').querySelector('button'),
+			action = status['_enabled'] ? 'stop' : 'start';
+
+		ddns_toggle.innerHTML = _(action.toUpperCase() + ' DDns')
+
+		L.dom.content(ddns_enabled, function() {
+			return E([], [
+				E('div', {}, status['_enabled'] ? _('DDNS Autostart enabled') : [
+					_('DDNS Autostart disabled'),
+					E('div', { 'class' : 'cbi-value-description' }, 
+					_("Currently DDNS updates are not started at boot or on interface events.") + "<br>" +
+					_("This is the default if you run DDNS scripts by yourself (i.e. via cron with force_interval set to '0')"))
+				]),]);
+		});
+			
+		for (var i = 0; i < rows.length; i++) {
+			section_id = rows[i].getAttribute('data-sid');
+			cfg_detail_ip = rows[i].querySelector('[data-name="_cfg_detail_ip"]');
+			cfg_update = rows[i].querySelector('[data-name="_cfg_update"]');
+			cfg_status = rows[i].querySelector('[data-name="_cfg_status"]');
+			reload = rows[i].querySelector('.cbi-section-actions .reload');
+			stop = rows[i].querySelector('.cbi-section-actions .stop');
+			cfg_enabled = uci.get('ddns', section_id, 'enabled');
+
+			reload.disabled = (status['_enabled'] == 0 || cfg_enabled == 0);
+			
+			host = uci.get('ddns', section_id, 'lookup_host') || _('Configuration Error');
+			ip =  _('No Data');
+			last_update = _('Never');
+			next_update = _('Unknown');
+			service_status = '<b>' + _('Not Running') + '</b>';
+			
+			if (service[section_id]) {
+				stop.disabled = (!service[section_id].pid);
+				if (service[section_id].ip) 
+					ip = service[section_id].ip;
+				if (service[section_id].last_update) 
+					last_update = _(service[section_id].last_update);
+				if (service[section_id].next_update) 
+					next_update = _(service[section_id].next_update);
+				if (service[section_id].pid)
+					service_status = '<b>' + _('Running') + '</b> : ' + service[section_id].pid;
+			}
+
+			cfg_detail_ip.innerHTML = host + '<br>' + ip;
+			cfg_update.innerHTML = last_update + '<br>' + next_update;
+			cfg_status.innerHTML = service_status;
+		}
+
+		return;
+	},
+
+	load: function() {
+		return Promise.all([
+			uci.load('ddns'),
+			this.callDDnsGetServicesStatus(),
+			callDDnsGetStatus(),
+			this.callDDnsGetEnv(),
+			L.resolveDefault(fs.read('/etc/ddns/services'), ''),
+			L.resolveDefault(fs.read('/etc/ddns/services_ipv6'), ''),
+		]);
+	},
+
+	render: function(data) {
+		var services = data[0];
+		var resolved = data[1];
+		var status = data[2];
+		var env = data[3];
+		var logdir = uci.get('ddns', 'global', 'ddns_logdir') || "/var/log/ddns";
+		
+		var services4 = []; 
+		var services6 = [];
+
+		data[4].split('\n').forEach(function(item) {
```

Comment:
You could consider replacing `L.resolveDefault(fs.read('/etc/ddns/services'), '')` with `fs.lines('/etc/ddns/services')`. The [`fs.lines`](http://openwrt.github.io/luci/jsapi/LuCI.fs.html#lines) function already splits and trims the file contents, and takes care of `resolveDefault()` internally to avoid errors.